tests/OgcApi.Net.Styles.Tests/FileSystemStorages/StyleFileSystemStorageTests.cs

```diff
@@ -0,0 +1,237 @@
+        const string expectedHref = "http://localhost/api/ogc/collections/testCollection/styles/existingStyleId?f=mapbox";
+
+        var style = await _styleFileSystemStorage.GetStyle(collectionId, styleId);
+        Assert.NotNull(style);
+        Assert.Equal(expectedTitle, style.Title);
+        Assert.Single(style.Links);
+        Assert.Equal(expectedHref, style.Links.First().Href.ToString());
+    }
+
+    [Fact]
+    public async Task GetStyles_ShouldReturnStyles()
+    {
+        const string collectionId = FileSystemFixture.CollectionId;
+        
+        var styles = await _styleFileSystemStorage.GetStyles(collectionId);
+
+        Assert.NotNull(styles);
+        Assert.Single(styles.Styles);
+        Assert.Null(styles.Default);
+    }
+
+    [Fact]
+    public async Task GetStylesheet_ShouldGetStylesheetContent()
+    {
+        const string collectionId = FileSystemFixture.CollectionId;
+        const string styleId = FileSystemFixture.ExistingStyleId;
+        const string format = "mapbox";
+        const string expectedContent = "style content";
+
+        var stylesheet = await _styleFileSystemStorage.GetStylesheet(collectionId, styleId, format);
+        Assert.NotNull(stylesheet);
+        Assert.Equal(expectedContent, stylesheet);
+    }
+
+    [Fact]
+    public async Task ReplaceStyle_ShouldReplaceStylesheetContent()
+    {
+        const string collectionId = FileSystemFixture.CollectionId;
+        const string styleId = FileSystemFixture.ExistingStyleId;
+        const string format = "mapbox";
+        const string expectedContent = "Completely new content";
+
+        var addStyleParameters = new StylesheetAddParameters
+        {
+            StyleId = styleId,
+            Format = format,
+            Content = expectedContent
+        };
+        await _styleFileSystemStorage.ReplaceStyle(collectionId, styleId, addStyleParameters);
+
+        var stylePath = Path.Combine(_options.BaseDirectory, collectionId, styleId, "style.mapbox.json");
+        var newContent = File.ReadAllText(stylePath);
+        Assert.Equal(expectedContent, newContent);
+    }
+
+    [Fact]
+    public async Task UpdateDefaultStyle_ShouldUpdateDefaultStyle()
+    {
+        const string collectionId = FileSystemFixture.CollectionId;
+        const string styleId = FileSystemFixture.ExistingStyleId;
+
+        var stylesInfo = await _styleFileSystemStorage.GetStyles(collectionId);
+        Assert.Null(stylesInfo.Default);
+
+        var defaultStyle = new DefaultStyle
+        {
+            Default = styleId
+        };
+        await _styleFileSystemStorage.UpdateDefaultStyle(collectionId, defaultStyle);
+
+        stylesInfo = await _styleFileSystemStorage.GetStyles(collectionId);
+        Assert.Equal(styleId, stylesInfo.Default);
+    }
+
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    protected virtual void Dispose(bool disposing)
```

Comment:
I think it's better to mark the class as sealed and remove the virtual overload of the Dispose method.